website/content/docs/job-specification/template.mdx

```diff
@@ -278,6 +278,38 @@ task "task" {
 }
 ```
 
+## Nomad Integration
+
+### Nomad Services
+
+Nomad service registrations can be queried using the `nomadService` and
+`nomadServices` functions. The requests are tied to the same namespace as the
+job which contains the template stanza.
+
+```hcl
+  template {
+    data = <<EOF
+# Configuration for a single NGINX upstream service.
+upstream my_app {
+  {{- range nomadService "my-app" }}
+  server {{ .Address }}:{{ .Port }};{{- end }}
```

Comment:
Should these ends be on this line? I notice in the Consul example they're on the next line.

(this one and line 304)

Reply:
Oh nvm, I see you changed it below too - I figure thats for a good reason

Reply:
It tidies up the rendered template formatting.

Previously it would look like:
```
upstream my_app {

  server 127.0.0.1:21836;
  
  server 127.0.0.1:29977;
  
}
```

Where now we get:
```
upstream my_app {
  server 127.0.0.1:21836;
  server 127.0.0.1:29977;
}
```